src/client.rs

```diff
@@ -7,37 +7,20 @@ use crate::{HttpClient, Request, RequestBuilder, Response, Result};
 
 use futures_util::future::BoxFuture;
 
-#[cfg(all(
-    feature = "curl-client",
-    not(any(feature = "h1-client", feature = "wasm-client"))
-))]
-use http_client::isahc::IsahcClient;
-
-#[cfg(all(
-    feature = "h1-client",
-    not(any(feature = "curl-client", feature = "wasm-client"))
-))]
-use http_client::h1::H1Client;
-
-#[cfg(any(
-    all(
-        feature = "wasm-client",
-        not(any(feature = "h1-client", feature = "curl-client"))
-    ),
-    feature = "wasm_bindgen"
-))]
-use http_client::wasm::WasmClient;
-
-#[cfg(all(
-    feature = "curl-client",
-    not(any(feature = "h1-client", feature = "wasm-client"))
-))]
+#[cfg(feature = "curl-client")]
+use http_client::isahc::IsahcClient as DefaultClient;
+
+#[cfg(feature = "wasm-client")]
```

Comment:
```suggestion
#[cfg(any(feature = "wasm-client", feature = "wasm_bindgen")]
```

The other lines should be `all(feature = "x", not(feature = "wasm_bindgen"))`.

Reply:
Wasm_bindgen enables wasm-client, though

Reply:
why would anyone ever enable both wasm and one of the other clients? that doesn't make sense, so we don't need to test for that condition throughout the code

Reply:
That is unfortunately how cargo features work, though.

```toml
surf = { version = "2", features = ["wasm-client"] }
```
Enables both the defaults (including `curl-client`) and `wasm-client`.

This was an oversight in my last `cfg` adjusting patch, adn https://github.com/http-rs/surf/pull/234 fixes that properly.

Reply:
yeah, people need to say default-features = false, and we need to document that. or we could not choose a default. wasm cannot build if other clients are enabled

Reply:
I do not think I agree. The `cfg` statements _can_ handle this fallback properly. I went that way originally but I can't see any downside to having the default client deal with feature overrides of the default.

Reply:
if the other features are enabled, the dependencies get pulled in. this is not what users want, but more importantly, those deps cannot build on wasm. people need to disable default features in order to use wasm (and i think we should encourage people to disable default features to switch to h1, as well, in order to not bloat their deps with unused crates)

Reply:
Hmm, true. Ok.